tests/test_rest_client.py

```
@@ -0,0 +1,1102 @@
+            await client.get_sleep_records()
+
+
+@pytest.mark.asyncio
+async def test_get_sleep_records_not_a_list(config_na, session):
+    client = RESTClient(config_na, session)
+    client._access_token = "access"
+    client._country_code = "US"
+    # 'items' is not a list
+    mock_response = {"data": {"getPatientWrapper": {"sleepRecords": {"items": "notalist"}}}}
+    with patch.object(client, "_gql_query", AsyncMock(return_value=mock_response)):
+        with pytest.raises(ParsingError, match="Returned records is not a list"):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)